File: scala/19.scala
import java.util.Calendar
import java.util.Calendar._

var cal = Calendar.getInstance
var sundays = 0

for (
    year <- 1901 to 2000;
    month <- 0 to 11) {
  cal.set(year, month, 1)
  if (cal.get(DAY_OF_WEEK) == SUNDAY) sundays += 1
}

val result = sundays

println(result)
assert(result == 171)

File: clojure/19.clj
(import [java.util Calendar])

(def cal (Calendar/getInstance))

(def sundays (atom 0))

(doseq [year (range 1901 2001)
        month (range 0 12)]
  (.set cal year month 1)
  (when (= (.get cal Calendar/DAY_OF_WEEK) Calendar/SUNDAY)
    (swap! sundays inc)))

(def result @sundays)

(println result)
(assert (= result 171))
